File: schemaorg-rs/src/opening_hours_specification.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///<https://schema.org/validThrough>
///<https://schema.org/Date>
///<https://schema.org/DateTime>
pub type OpeningHoursSpecificationValidThroughFieldEnum = String;
///<https://schema.org/additionalType>
///<https://schema.org/Text>
///<https://schema.org/URL>
pub type OpeningHoursSpecificationAdditionalTypeFieldEnum = String;
///<https://schema.org/validFrom>
///<https://schema.org/Date>
///<https://schema.org/DateTime>
pub type OpeningHoursSpecificationValidFromFieldEnum = String;
///<https://schema.org/OpeningHoursSpecification>
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct OpeningHoursSpecification {
    #[serde(rename = "@context")]
    pub context: String,
    ///<https://schema.org/image>
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<OpeningHoursSpecificationImageFieldEnum>,
    ///<https://schema.org/name>
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///<https://schema.org/validThrough>
    #[serde(rename = "validThrough")]
    #[serde_as(as = "OneOrMany<_>")]
    pub valid_through: Vec<OpeningHoursSpecificationValidThroughFieldEnum>,
    ///<https://schema.org/opens>
    #[serde(rename = "opens")]
    #[serde_as(as = "OneOrMany<_>")]
    pub opens: Vec<String>,
    ///<https://schema.org/subjectOf>
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<OpeningHoursSpecificationSubjectOfFieldEnum>,
    ///<https://schema.org/potentialAction>
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///<https://schema.org/additionalType>
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<OpeningHoursSpecificationAdditionalTypeFieldEnum>,
    ///<https://schema.org/description>
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<OpeningHoursSpecificationDescriptionFieldEnum>,
    ///<https://schema.org/url>
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///<https://schema.org/identifier>
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<OpeningHoursSpecificationIdentifierFieldEnum>,
    ///<https://schema.org/alternateName>
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///<https://schema.org/closes>
    #[serde(rename = "closes")]
    #[serde_as(as = "OneOrMany<_>")]
    pub closes: Vec<String>,
    ///<https://schema.org/validFrom>
    #[serde(rename = "validFrom")]
    #[serde_as(as = "OneOrMany<_>")]
    pub valid_from: Vec<OpeningHoursSpecificationValidFromFieldEnum>,
    ///<https://schema.org/mainEntityOfPage>
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<OpeningHoursSpecificationMainEntityOfPageFieldEnum>,
    ///<https://schema.org/disambiguatingDescription>
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///<https://schema.org/dayOfWeek>
    #[serde(rename = "dayOfWeek")]
    #[serde_as(as = "OneOrMany<_>")]
    pub day_of_week: Vec<DayOfWeekEnum>,
    ///<https://schema.org/sameAs>
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
}

File: schemaorg-ts/src/opening_hours_specification.ts
import type * as all from "./index";

/**
 * <https://schema.org/validFrom>
 **/
export type OpeningHoursSpecificationValidFromFieldEnum = string;

/**
 * <https://schema.org/validThrough>
 **/
export type OpeningHoursSpecificationValidThroughFieldEnum = string;

/**
 * <https://schema.org/additionalType>
 **/
export type OpeningHoursSpecificationAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/OpeningHoursSpecification>
 **/
export interface OpeningHoursSpecification {
  "@context": string;
  "@type": "OpeningHoursSpecification";

  /**
   * <https://schema.org/closes>
   **/
  closes: all.OneOrMany<string>;

  /**
   * <https://schema.org/validFrom>
   **/
  validFrom: all.OneOrMany<all.OpeningHoursSpecificationValidFromFieldEnum>;

  /**
   * <https://schema.org/opens>
   **/
  opens: all.OneOrMany<string>;

  /**
   * <https://schema.org/validThrough>
   **/
  validThrough: all.OneOrMany<all.OpeningHoursSpecificationValidThroughFieldEnum>;

  /**
   * <https://schema.org/dayOfWeek>
   **/
  dayOfWeek: all.OneOrMany<all.DayOfWeekEnum>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.OpeningHoursSpecificationMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.OpeningHoursSpecificationIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.OpeningHoursSpecificationDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.OpeningHoursSpecificationSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.OpeningHoursSpecificationAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.OpeningHoursSpecificationImageFieldEnum>;
}
